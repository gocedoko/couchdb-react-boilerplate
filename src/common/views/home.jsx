import { React, withStyles, connect, _t, Paper, Typography } from "../../imports.jsx"
import styles from "../style.jsx"

export const _Home = props =>
	<main className={props.classes.main}>
		<Paper 
			className={props.classes.paper}>
				<Typography 
					component="h1" 
					variant="h5" 
					gutterBottom> 
						{`${_t('Welcome')} ${props.signedInUserData.firstName} ${props.signedInUserData.lastName}`} 
				</Typography>
						
				<Typography 
					component="p" 
					variant="body2"> 
						{_t('This is the home page of the couchdb-react boilerplate preview')}.
				</Typography>
		</Paper>
	</main>


export const Home = connect(
	state => ({
		signedInUserData: state.signedInUserData,
	})
)(withStyles(styles)(_Home))
