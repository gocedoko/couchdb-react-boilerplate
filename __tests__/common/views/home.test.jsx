import { React, withStyles, Provider, shallow, mount, render, toJson, configureStore 
      } from '../../../src/imports.jsx'

import { _Home } from "../../../src/common/views/home.jsx"
import styles from "../../../src/common/style.jsx"

describe('<Home />', () => {
  describe('render() with empty state', () => {
    test('component is rendered exactly like the last time', () => 
    {
        const HomeInstance = withStyles(styles)(props => 
          <_Home 
            {...props}
            signedInUserData={{}}
          />)
        expect(toJson(render(<HomeInstance/>))).toMatchSnapshot()
    })
  })

  describe('render() with user details', () => {
    test('component is rendered exactly like the last time', () => 
    {
        const HomeInstance = withStyles(styles)(props => 
          <_Home 
              {...props}
              signedInUserData= {{  
                firstName: "Goce",
                lastName: "Dokoski"
              }}
            />)
        expect(toJson(render(<HomeInstance/>))).toMatchSnapshot()
    })
  })
})