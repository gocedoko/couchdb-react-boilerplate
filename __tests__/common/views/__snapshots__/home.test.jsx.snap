// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render() with empty state component is rendered exactly like the last time 1`] = `
<main>
  <div
    class="MuiPaper-root-18 MuiPaper-elevation2-22 MuiPaper-rounded-19 Component-paper-17"
  >
    <h1
      class="MuiTypography-root-45 MuiTypography-h5-61 MuiTypography-gutterBottom-72"
    >
      Welcome undefined undefined
    </h1>
    <p
      class="MuiTypography-root-45 MuiTypography-body2-53"
    >
      This is the home page of the couchdb-react boilerplate preview.
    </p>
  </div>
</main>
`;

exports[`<Home /> render() with user details component is rendered exactly like the last time 1`] = `
<main>
  <div
    class="MuiPaper-root-18 MuiPaper-elevation2-22 MuiPaper-rounded-19 Component-paper-97"
  >
    <h1
      class="MuiTypography-root-45 MuiTypography-h5-61 MuiTypography-gutterBottom-72"
    >
      Welcome Goce Dokoski
    </h1>
    <p
      class="MuiTypography-root-45 MuiTypography-body2-53"
    >
      This is the home page of the couchdb-react boilerplate preview.
    </p>
  </div>
</main>
`;
